src/python/dart/engine/emr/actions/start_datastore.py

```diff
@@ -174,3 +179,53 @@ def prepare_bootstrap_actions(datastore, impala_docker_repo_base_url, impala_ver
         s3_copy_recursive(local_ba_path, s3_ba_path)
 
     return results
+
+
+def prepare_cluster_configurations(configuration_overrides=None):
+    configs_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'start_configs.json')
+    if configuration_overrides:
+        with open(configs_path, 'rt') as in_file:
+            configs = json.load(in_file)
+        merge_configurations(configuration_overrides, configs)
+        fd, configs_path = tempfile.mkstemp(suffix='.json', prefix='configs', text=True)
+        with os.fdopen(fd, 'wt') as out_file:
+            json.dump(configs, out_file, indent=True)
+    return 'file://%s' % configs_path
+
+
+def merge_configurations(source, target):
+    """
+    Merge two EMR configurations arrays consisting of configuration dictionaries, merging
+    elements with the same Classification identifier with source overwriting target.
+
+    See http://docs.aws.amazon.com/ElasticMapReduce/latest/ReleaseGuide/emr-configure-apps.html
+
+    :param source Source configurations array
+    :param target Target configurations array
+    """
+    target_config_by_classification = dict([(config['Classification'], config) for config in target])
```

Comment:
Alternatively, this could have also been written with dictionary comprehension:
`{config['Classification']: config for config in target}`